Sources/MapboxDirections/RouteStep.swift

```diff
@@ -534,7 +541,35 @@ open class RouteStep: Codable {
         }
     }
     
-    public required init(from decoder: Decoder) throws {
+    static func decode(from decoder: Decoder, administrativeRegions: [AdministrativeRegion]) throws -> [RouteStep] {
```

Comment:
`init(from:)` is required for Decodable conformance. The administrative regions need to be passed in via the decoder’s `userInfo`, just like `routeOptions`.

Reply:
`userInfo` is not mutable during decoding process, so admin regions should be configured by user, outside.
About updating entities after the actual decoding - I didn't want to create series of `mutating` methods, as `Intersection` is a struct.

Reply:
If we need to define this custom method, then we still need an explicit `init(from:)`; otherwise, Codable will synthesize one for us that may not match the expected structure.

Reply:
explicit `init(from:)` is still [available](https://github.com/mapbox/mapbox-directions-swift/pull/485/files#diff-6f755dc6c64528357f03e87f359b0f8c20d9c827b3cedf8ff27eec0c472542d0R568-R570) and both methods share decoding logic.